src/sql/mod.rs

```diff
@@ -113,20 +114,35 @@ impl SqlBuilder {
         }
     }
 
-    pub(crate) fn finish(mut self) -> Result<String> {
+    pub(crate) fn finish(mut self, interp_flags: u8) -> Result<String> {
         let mut sql = String::new();
 
         if let Self::InProgress(parts, _) = &self {
             for part in parts {
                 match part {
                     Part::Text(text) => sql.push_str(text),
                     Part::Arg => {
-                        self.error("unbound query argument");
-                        break;
+                        if QI::has_bind(interp_flags) {
+                            self.error("unbound query argument");
+                            break;
+                        } else {
+                            // Push NULL as placeholder to prevent ClickHouse server error
+                            eprintln!(
+                                "warning use QI::BIND template flag,bind() skipped for query: {}",
+                                self
+                            );
```

Comment:
It's better to use the log crate here.

```suggestion
                            log::warn!("use QI::BIND template flag,bind() skipped for query: {self}");
```

Reply:
Thanks, I will fix it.